src/CSTools/Tools/lexer.cs

```diff
@@ -970,21 +972,23 @@ public ErrorHandler() {}
 		public virtual void Error(Exception e)
 		{
 			counter++;
-			if (throwExceptions || (e is CSToolsFatalException))
```

Comment:
we probably still need this as `Error` could be called with a `CSToolsFatalException` but if the caller doesn't know the type then the new method `Error(CSToolsFatalException e)` won't be called.

Reply:
You're right that there's a problem, but I was thrown off when you mentioned the caller not knowing the type. Whether they know the type or not doesn't matter (I don't think), what matters is that C# resolves its methods at compile-time. So the problem with this code is that 

				CSToolsException e = new CSToolsFatalException(...);
				syms.erh.Error(e);

calls the less derived method. Or maybe there's another problem?

Reply:
Yes that's what I was referring to. In this case the 2nd line doesn't know the actual type, just that it's a tools exception.

Reply:
Right, I get what you're saying now!